src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -927,14 +929,14 @@ private void CreateResponseHeader(bool appCompleted)
                         //
                         // A server MUST NOT send a response containing Transfer-Encoding unless the corresponding
                         // request indicates HTTP/1.1 (or later).
-                        if (_httpVersion == Http.HttpVersion.Http11 && StatusCode != StatusCodes.Status101SwitchingProtocols)
+                        if (statusCode != StatusCodes.Status101SwitchingProtocols && httpVersion == Http.HttpVersion.Http11)
```

Comment:
Switched these originally as made `statusCode` register but hadn't done `_httpVersion` (wasn't sure if would run out of registers); so made it a lazy check. Switch not needed now...